app/services/school_partnerships/query.rb

```diff
@@ -1,72 +1,101 @@
 module SchoolPartnerships
   class Query
+    include Queries::ConditionFormats
+    include Queries::Orderable
+    include Queries::FilterIgnorable
+
     attr_reader :scope
 
-    def initialize(school: :ignore, contract_period: :ignore, lead_provider: :ignore, delivery_partner: :ignore)
+    def initialize(school_id: :ignore, contract_period_years: :ignore, lead_provider_id: :ignore, delivery_partner_api_ids: :ignore, updated_since: :ignore, sort: nil)
       @scope = default_scope
 
-      where_lead_provider(lead_provider)
-      where_contract_period(contract_period)
-      where_school(school)
-      where_delivery_partner(delivery_partner)
+      where_lead_provider_is(lead_provider_id)
+      where_contract_period_year_in(contract_period_years)
+      where_school_is(school_id)
+      where_delivery_partner_is(delivery_partner_api_ids)
+      where_updated_since(updated_since)
+      set_sort_by(sort)
     end
 
     delegate(:exists?, to: :scope)
 
     def school_partnerships
-      scope.earliest_first
+      scope
     end
 
     def earliest_school_partnership
       school_partnerships.first
     end
 
+    def school_partnership_by_api_id(api_id)
+      return scope.find_by!(api_id:) if api_id.present?
+
+      fail(ArgumentError, "You must specify an api_id")
```

Comment:
I think for consistency let's use `api_id needed` same below, although maybe worth fixing those generally in the others later as this is a nicer error message